cpp/src/arrow/util/binary_view_util.h

```diff
@@ -26,6 +26,7 @@
 namespace arrow::util {
 
 inline BinaryViewType::c_type ToInlineBinaryView(const void* data, int32_t size) {
+  assert(size <= BinaryViewType::kInlineSize);
```

Comment:
https://godbolt.org/z/q8xjb8Ehe 
From here, COMPILER ASSUMER might generate better code in C++

Reply:
Which I think we be assumed when this is inlined from the `if` below.